internal/provider/resource_time_rotating.go

```diff
@@ -0,0 +1,710 @@
+	RotationDays    types.Int64  `tfsdk:"rotation_days"`
+	RotationHours   types.Int64  `tfsdk:"rotation_hours"`
+	RotationMinutes types.Int64  `tfsdk:"rotation_minutes"`
+	RotationMonths  types.Int64  `tfsdk:"rotation_months"`
+	RotationRFC3339 types.String `tfsdk:"rotation_rfc3339"`
+	RotationYears   types.Int64  `tfsdk:"rotation_years"`
+	Hour            types.Int64  `tfsdk:"hour"`
+	Triggers        types.Map    `tfsdk:"triggers"`
+	Minute          types.Int64  `tfsdk:"minute"`
+	Month           types.Int64  `tfsdk:"month"`
+	RFC3339         types.String `tfsdk:"rfc3339"`
+	Second          types.Int64  `tfsdk:"second"`
+	Unix            types.Int64  `tfsdk:"unix"`
+	Year            types.Int64  `tfsdk:"year"`
+	ID              types.String `tfsdk:"id"`
+}
+
+func setRotationValues(plan *timeRotatingModelV0, timestamp time.Time) (timeRotatingModelV0, error) {
+	formattedTimestamp := timestamp.Format(time.RFC3339)
+
+	var rotationTimestamp time.Time
+
+	if plan.RotationDays.Value != 0 {
+		rotationTimestamp = timestamp.AddDate(0, 0, int(plan.RotationDays.Value))
+	}
+
+	if plan.RotationHours.Value != 0 {
+		hours := time.Duration(plan.RotationHours.Value) * time.Hour
+		rotationTimestamp = timestamp.Add(hours)
+	}
+
+	if plan.RotationMinutes.Value != 0 {
+		minutes := time.Duration(plan.RotationMinutes.Value) * time.Minute
+		rotationTimestamp = timestamp.Add(minutes)
+	}
+
+	if plan.RotationMonths.Value != 0 {
+		rotationTimestamp = timestamp.AddDate(0, int(plan.RotationMonths.Value), 0)
+	}
+
+	if plan.RotationRFC3339.Value != "" {
+		var err error
+
+		if rotationTimestamp, err = time.Parse(time.RFC3339, plan.RotationRFC3339.Value); err != nil {
+			return timeRotatingModelV0{}, err
+		}
+	}
+
+	if plan.RotationYears.Value != 0 {
+		rotationTimestamp = timestamp.AddDate(int(plan.RotationYears.Value), 0, 0)
+	}
+
+	formattedRotationTimestamp := rotationTimestamp.Format(time.RFC3339)
+
+	return timeRotatingModelV0{
+		RotationRFC3339: types.String{Value: formattedRotationTimestamp},
+		Triggers:        plan.Triggers,
+		Year:            types.Int64{Value: int64(rotationTimestamp.Year())},
+		Month:           types.Int64{Value: int64(rotationTimestamp.Month())},
+		Day:             types.Int64{Value: int64(rotationTimestamp.Day())},
+		Hour:            types.Int64{Value: int64(rotationTimestamp.Hour())},
+		Minute:          types.Int64{Value: int64(rotationTimestamp.Minute())},
+		Second:          types.Int64{Value: int64(rotationTimestamp.Second())},
+		RotationYears:   plan.RotationYears,
+		RotationMonths:  plan.RotationMonths,
+		RotationDays:    plan.RotationDays,
+		RotationHours:   plan.RotationHours,
+		RotationMinutes: plan.RotationMinutes,
+		RFC3339:         types.String{Value: formattedTimestamp},
+		Unix:            types.Int64{Value: rotationTimestamp.Unix()},
+		ID:              types.String{Value: formattedTimestamp},
+	}, nil
+
+}
+
+func parseTwoPartId(idParts []string, resp *resource.ImportStateResponse) timeRotatingModelV0 {
```

Comment:
Consider removing `resp *resource.ImportStateResponse` as argument and returning `(timeRotatingModelV0, error)`.